webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -21,18 +21,22 @@ export const getNotifications = url => dispatch => {
       document.visibilityState === 'visible' ||
       document.visibilityState === 'prerender'
     ) {
-      API.get(url).then(
-        response => {
+      API.get(url)
+        .success(response => {
           dispatch({
             type: NOTIFICATIONS_GET_NOTIFICATIONS,
             payload: {
               notifications: response.notifications
             }
           });
           resolve();
-        },
-        () => reject()
-      );
+        })
+        .fail(response => {
+          if (response.status === 401) {
+            window.location.replace('/users/login');
```

Comment:
in notifications.test.js , we already have mock server response in the beforeEach part, we need to mock a failure response, and then set up a mock for window.location.replace function with a spy, and make sure that it is called.